### ISSUE TITLE
Decode object showing up nil

### ISSUE THREAD

**Issue by kbsrunschi** [uchicago-mobi/2016-Summer-Forum#31]
let name = aDecoder.decodeObjectForKey(PropertKey.nameKey) as! String

I ran my code and the app worked correctly. I saved, committed the code, and pushed, and now I am getting a nil exception for the above line. I compared my code against the Module code and everything matches. 

Prior to this error, I received an error saying there was a duplication of encoding keys in the class hierarchy, but the project ran with no issues.

**Comment by susanstevens** [comment]
You might need to delete the app in the simulator and do a fresh build.  Command-shift-H will take you to the simulator's home screen.  Then click and hold on the app icon until it starts wiggling and click the x to delete.

If you changed the keys used to encode and decode objects at any point, the app might be trying to decode using a key that was never used to encode anything.  Make sure you run the app twice after you rebuild.  Add some new meals, then stop and _close_ the app before running again to verify that the meals were saved.  (If you don't actually close the app in the simulator, you won't be verifying that data is saved between sessions.)

**Comment by kbsrunschi** [comment]
So after deleting the app in the simulator and cleaning and rebuilding, I was able to run my app and edit/delete/save. I was still getting the duplication of encoding keys error.

I stopped the app and closed the simulator and reran, and then I got the "found nil while unwrapping" and the app no longer builds so ill investigate that path. It is saving info between sessions as I was able to see a previously deleted meal item remains deleted in the new session.
